en/components/tree.md

```diff
@@ -0,0 +1,259 @@
+
+```html
+<igx-tree>
+	<igx-tree-node *ngFor="let nodes of data" [data]="node" [expanded]="isNodeExpaded(node)" [selected]="isNodeSelected(node)">
+		{{ node.text }}
+		<img [src]="node.image" alt="node.imageAlt" />
+		<igx-tree-node *ngFor="let child of node.children" [data]="child">
+				<a igxTreeNodeLink href="child.url" target="_blank">
+                    {{ child.text }}
+                </a>
+		</igx-tree-node>
+	</igx-tree-node>
+</igx-tree>
+```
+### Node Interactions
+IgxTree provides the following API methods for node interactions:
+- [**expand**]({environment:angularApiUrl}/classes/igxtreenodecomponent.html#expand) - expands the node
+- [**collapse**]({environment:angularApiUrl}/classes/igxtreenodecomponent.html#collapse) - collapses the node
+- [**toggle**]({environment:angularApiUrl}/classes/igxtreenodecomponent.html#toggle) - toggles node expansion state
+- [**collapseAll**]({environment:angularApiUrl}/classes/igxtreecomponent.html#collapseAll) - collapses the specified nodes. If no nodes are passed, collapses all parent nodes.
+- [**expandAll**]({environment:angularApiUrl}/classes/igxtreecomponent.html#expandAll) - sets the specified nodes as expanded. If no nodes are passed, expands all parent nodes.
+- [**deselectAll**]({environment:angularApiUrl}/classes/igxtreecomponent.html#deselectAll) - deselects all nodes. If a nodes array is passed, deselects only the specified nodes. Does not emit nodeSelection event.
+
+### Finding Nodes
+You can find a specific node within IgxTree by using the [findNodes]({environment:angularApiUrl}/classes/igxtreecomponent.html#findNodes) method.  It returns an array of nodes, which match the specified data.
+When finding nodes, you can pass a custom comparer function in order to find the data.
+```html
+<igx-tree>
+	<igx-tree-node *ngFor="let nodes of data" [data]="node" [expanded]="isNodeExpaded(node)" [selected]="isNodeSelected(node)">
+		{{ node.text }}
+		<img [src]="node.image" alt="node.imageAlt" />
+		<igx-tree-node *ngFor="let child of node.children" [data]="child" [expanded]="isNodeExpaded(child)" [selected]="isNodeSelected(child)">
+      {{ child.text }}
+		</igx-tree-node>
+	</igx-tree-node>
+</igx-tree>
+```
+```typescript
+export class MyTreeViewComponent {
+  public data: { [key: string]: any, valueKey: string } = MY_DATA;
+  @ViewChild(IgxTreeComponent, { read: IgxTreeComponent })
+  public tree;
+
+  findNode(valueKey: string) {
+    const comparer: (nodeData: any, node: IgxTreeNodeComponent) => boolean =
+      (data: any, node: IgxTreeNodeComponent) => nodeData.valueKey === data;
+    this.tree.findData(valueKey, comparer);
+  }
+}
+```
+### Templating
+To create a custom node template within igx-tree, declare `<ng-template>`. From the template you can access the implicitly provided context variable, with properties that give you information about the node. 
```

Comment:
Emphasis on the part where the template needs to be within the tree ('cause if someone puts a `igxTreeNodeLink` directive, it has to have a provider for tree). The seconds sentence is applicable for any templates, should be omitted IMO
```suggestion
To create a reusable template for your nodes, declare `<ng-template>`  **within `igx-tree`**. 
```